classes/doi/Repository.inc.php

```diff
@@ -0,0 +1,370 @@
+     *
+     *
+     */
+    public function depositAll(Context $context)
+    {
+        $enabledDoiTypes = $context->getData(Context::SETTING_ENABLED_DOI_TYPES);
+        if ($this->_checkIfSubmissionValidForDeposit($enabledDoiTypes)) {
+
+            // If there is no configured registration agency, nothing can be deposited.
+            $agency = $this->_getAgencyFromContext($context);
+            if (!$agency) {
+                return;
+            }
+
+            $submissionsCollection = $this->dao->getAllDepositableSubmissionIds($context);
+            $submissionData = $submissionsCollection->reduce(function ($carry, $item) {
+                if ($item->submission_id) {
+                    $carry['submissionIds'][] = $item->submission_id;
+                }
+                $carry['doiIds'][] = $item->doi_id;
+
+                return $carry;
+            }, ['submissionIds' => [], 'doiIds' => []]);
+
+            // Schedule/queue jobs for submissions
+            foreach ($submissionData['submissionIds'] as $submissionId) {
+                dispatch(new DepositSubmission($submissionId, $context, $agency));
+            }
+
+            // Mark submission DOIs as submitted
+            Repo::doi()->markSubmitted($submissionData['doiIds']);
+        }
+    }
+
+    /**
+     * Loops over valid submission DOI types to see if any are enabled
+     *
+     */
+    private function _checkIfSubmissionValidForDeposit(array $enabledDoiTypes): bool
+    {
+        foreach ($this->getValidSubmissionDoiTypes() as $validSubmissionDoiType) {
+            if (in_array($validSubmissionDoiType, $enabledDoiTypes)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Get app-specific DOI type constants to check when scheduling deposit for submissions
+     *
+     */
+    abstract protected function getValidSubmissionDoiTypes(): array;
+
+    /**
+     * Gets all relevant DOI IDs related to a submission
+     * NB: Assumes current publication only and only enabled DOI types
+     *
+     * @return array DOI IDs
+     */
+    abstract public function getDoisForSubmission(int $submissionId): array;
+
+    /**
+     * Compose final DOI and save to database
+     *
+     */
+    protected function mintAndStoreDoi(Context $context, string $doiSuffix): ?int
+    {
+        $doiPrefix = $context->getData(Context::SETTING_DOI_PREFIX);
+        if (empty($doiPrefix)) {
+            return null;
+        }
+
+        $completedDoi = $doiPrefix . '/' . $doiSuffix;
+
+        $doiDataParams = [
+            'doi' => $completedDoi,
+            'contextId' => $context->getId()
+        ];
+
+        $doi = $this->newDataObject($doiDataParams);
+        return $this->add($doi);
+    }
+
+    /**
+     * Helper to retrieve and confirm validity of registration agency for a given context
+     */
+    protected function _getAgencyFromContext(Context $context): ?IDoiRegistrationAgency
```

Comment:
This function, `_getAgencyFromContext`, seems obsolete now that you've implemented `Context::getConfiguredDoiAgency()`. In both cases, the agency is fetched, checked to see if it is enabled, and checked to see if it fits the right interface.

I think you could remove this method, as well as the other one on the pkp-lib repository, and just use `$context->getConfiguredDoiAgency()` directly.